src/common/services/api/cart.service.js

```diff
@@ -162,8 +163,10 @@ class Cart {
   }
 
   editItem (oldUri, uri, data) {
-    return this.deleteItem(oldUri)
-      .switchMap(() => this.addItem(uri, data, true))
+    return Observable.concat(
+      this.deleteItem(oldUri),
+      this.addItem(uri, data, true)
```

Comment:
The ```concat``` just prevents ```this.addItem``` from starting until ```this.deleteItem``` finishes right?

Reply:
That's my understanding. See the [docs](https://www.learnrxjs.io/learn-rxjs/operators/combination/concat).